src/plugins/notebook/components/Notebook.vue

```diff
@@ -377,25 +402,66 @@ export default {
             const output = [];
             const entries = this.internalDomainObject.configuration.entries;
             const sectionKeys = Object.keys(entries);
+            const searchTextLower = this.search.toLowerCase();
+            const originalSearchText = this.search;
+
             sectionKeys.forEach(sectionKey => {
                 const pages = entries[sectionKey];
                 const pageKeys = Object.keys(pages);
+                const section = this.getSection(sectionKey);
+                const sectionHit = section.name && section.name.toLowerCase().includes(searchTextLower);
+
                 pageKeys.forEach(pageKey => {
                     const pageEntries = entries[sectionKey][pageKey];
+                    const page = this.getPage(section, pageKey);
+                    const pageHit = page.name && page.name.toLowerCase().includes(searchTextLower);
+
                     pageEntries.forEach(entry => {
-                        if (entry.text && entry.text.toLowerCase().includes(this.search.toLowerCase())) {
-                            const section = this.getSection(sectionKey);
-                            output.push({
-                                section,
-                                page: this.getPage(section, pageKey),
-                                entry
-                            });
+                        const entryHit = entry.text && entry.text.toLowerCase().includes(searchTextLower);
+
+                        if (sectionHit || pageHit || entryHit) {
+                            const resultMetadata = {
+                                searchTextLower,
+                                originalSearchText,
+                                sectionHit,
+                                pageHit,
+                                entryHit
+                            };
+                            let resultPage = page;
+
+                            // if there is no page hit or entryHit (just section),
+                            // then we show first page by default
+                            if (sectionHit && !pageHit && !entryHit) {
+                                resultPage = this.getPage(section, pageKeys[0]);
+                            }
+
+                            // entryHits: (unique) go in
+                            // pageHits: with no entry (would be in alrady if entry hit) go in if the page isn't in already
+                            // sectionHits: no page or entry hits and section hit (everything else covered)
+                            if (
```

Comment:
I believe with checking at end logic we may not need this condition. 😄

Reply:
See above :)